components/engine_traits/src/raft_engine.rs

```diff
@@ -138,6 +140,10 @@ pub trait RaftEngine: RaftEngineReadOnly + PerfContextExt + Clone + Sync + Send
     where
         F: FnMut(u64) -> std::result::Result<(), E>,
         E: From<Error>;
+
+    fn recover_from_raftdb(&self) -> Result<bool>;
+
+    fn put_recover_from_raftdb(&self, recover_from_raftdb: bool) -> Result<()>;
```

Comment:
Add some comment to describe what's the purpose of the function. And it seems wrong to me as the key is named `store_version_key`, but the method name is `put_recover_from_raftdb`. What if `store_version` is assigned to another value? It can easily be overwritten by mistake.